packages/website/docs/NumericInput.md

```diff
@@ -1,17 +1,34 @@
-## Guidelines
+# Best Practices
 
--   The input field should be large enough to display all the user input in most cases.
--   The input field width should suggest the expected size of the user input.
--   Ensure that users with small screens can still enter all required information.
+Use a numeric input when the user needs **small and precise** incremental changes.
 
----
+A numeric input is more appropriate when the value is set within a shorter range. If your value has no limit, or if the limit is arbitrary, consider removing the incremental stepper.
 
-## Related Components
+The width of a numeric input should be large enough to display the maximum number of digits.
+
+## Labeling
+
+Keep labels short, preferably **under three words.**
+
+Provide a descriptive title without action verbs. For example, write "Price" rather than "Enter your price".
+
+## Help Text and Instructions
+
+If the input is set within a range, consider indicating the minimum and maximum values, or only one of them, in a help text, e.g. "Maximum of 20".
 
-[Text Input](#/components/Input)
+## Feedback and Validation
 
-[Radio Buttons](#/components/Radio)
+Numerical inputs are best when the value doesn't require decimal or fraction. If decimals are expected to be frequently set by users, make sure to show **the decimal point** in the default value, e.g.: "20.0", to indicate which format is accepted.
+
+Always shows in the validation message both the minimal and maximal limit if the user tries to enter a number outside the acceptable range.
+
+Determine the stepper increment relative to the range. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-100, the increments should be of 10.
```

Comment:
```suggestion
Determine the stepper increment relative to the range. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-100, the increments should be of 10.
```
@kpoulincoveo Not sure I understand the advice here. Can we be more specific? Should we recommend increments of 10%?

Reply:
I think we can suggest an increment of 10% but am not sure it always gonna be the case.